packages/p2p-v2/src/components/PaymentMethodForm/PaymentMethodForm.tsx

```diff
@@ -0,0 +1,240 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { TFormState } from 'src/reducers/types';
+import { TSelectedPaymentMethod } from 'types';
+import { p2p } from '@deriv/api';
+import { Button } from '@deriv-com/ui/dist/components/Button';
+import { Text } from '@deriv-com/ui/dist/components/Text';
+import CloseCircle from '../../public/ic-close-circle.svg';
+import { ClickableText } from '../ClickableText';
+import { Dropdown } from '../Dropdown';
+import { PaymentMethodErrorModal, PaymentMethodModal } from '../Modals';
+import { PaymentMethodField } from '../PaymentMethodField';
+import { PaymentMethodsHeader } from '../PaymentMethodsHeader';
+import { TextField } from '../TextField';
+import './PaymentMethodForm.scss';
+
+type TPaymentMethodFormProps = {
+    formState: TFormState;
+    onAdd: (selectedPaymentMethod?: TSelectedPaymentMethod) => void;
+    onResetFormState: () => void;
+};
+
+/**
+ * @component This component is used to display the form to add or edit a payment method
+ * @param formState - The current state of the form
+ * @returns {JSX.Element}
+ * @example <PaymentMethodForm formState={formState} />
+ * **/
+const PaymentMethodForm = ({ onAdd, onResetFormState, ...rest }: TPaymentMethodFormProps) => {
+    const {
+        control,
+        formState: { isDirty, isSubmitting, isValid },
+        handleSubmit,
+        reset,
+    } = useForm({ mode: 'all' });
+    const [isModalOpen, setIsModalOpen] = useState(false);
+    const { actionType, selectedPaymentMethod, title } = rest.formState || {};
+
+    const { data: availablePaymentMethods } = p2p.paymentMethods.useGet();
+    const { create, error: createError, isSuccess: isCreateSuccessful } = p2p.advertiserPaymentMethods.useCreate();
+    const { error: updateError, isSuccess: isUpdateSuccessful, update } = p2p.advertiserPaymentMethods.useUpdate();
+
+    useEffect(() => {
+        if (isCreateSuccessful) {
+            onResetFormState();
+        } else if (createError) {
+            setIsModalOpen(true);
+        }
+    }, [isCreateSuccessful, createError, onResetFormState]);
+
+    useEffect(() => {
+        if (isUpdateSuccessful) {
+            onResetFormState();
+        } else if (updateError) {
+            setIsModalOpen(true);
+        }
+    }, [isUpdateSuccessful, onResetFormState, updateError]);
```

Comment:
can we not club the two useeffects?

Reply:
@nada-deriv,  I believe each `effect` should represent a separate `side effect`. This would reduce the chances of unpredictable behaviour, incase someone is to alter any of the `effects` later on